File: src/test/scala/com/exasol/cloudetl/kafka/deserialization/FieldSpecificationTest.scala
package com.exasol.cloudetl.kafka.deserialization

import nl.jqno.equalsverifier.EqualsVerifier
import org.scalatest.funsuite.AnyFunSuite

class FieldSpecificationTest extends AnyFunSuite {

  test("equals and hashCode of record key field must follow the contract") {
    EqualsVerifier.forClass(classOf[RecordKeyField]).verify()
  }

  test("equals and hashCode of record value field must follow the contract") {
    EqualsVerifier.forClass(classOf[RecordValueField]).verify()
  }

  test("timestamp field must behave as a singleton for equals and hashCode") {
    assert(TimestampField === TimestampField)
    assert(TimestampField.hashCode === TimestampField.hashCode)
    assert(!TimestampField.equals(null))
  }
}

File: src/test/java/com/exasol/cloudetl/kafka/deserialization/FieldSpecificationTest.java
package com.exasol.cloudetl.kafka.deserialization;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

import nl.jqno.equalsverifier.EqualsVerifier;

class FieldSpecificationTest {
    @Test
    void recordKeyFieldEqualsAndHashCodeFollowContract() {
        EqualsVerifier.forClass(RecordKeyField.class).verify();
    }

    @Test
    void recordValueFieldEqualsAndHashCodeFollowContract() {
        EqualsVerifier.forClass(RecordValueField.class).verify();
    }

    @Test
    void timestampFieldBehavesAsSingletonForEqualsAndHashCode() {
        assertEquals(TimestampField.INSTANCE, TimestampField.INSTANCE);
        assertEquals(TimestampField.INSTANCE.hashCode(), TimestampField.INSTANCE.hashCode());
        assertNotEquals(null, TimestampField.INSTANCE);
    }
}
